seed/views/column_list_settings.py

```diff
@@ -46,3 +55,92 @@ class ColumnListingViewSet(OrgValidateMixin, SEEDOrgCreateUpdateModelViewSet):
     pagination_class = None
     # force_parent = True  # Ideally the column list settings would inherit from the parent,
     # but not yet.
+
+    # Overridden to augment with protected ComStock list setting if enabled
+    def retrieve(self, request, *args, **kwargs):
+        org_id = self.get_organization(self.request)
+
+        try:
+            org = Organization.objects.get(pk=org_id)
+        except Organization.DoesNotExist:
+            return JsonResponse({
+                'status': 'error',
+                'message': 'organization with id %s does not exist' % org_id
+            }, status=status.HTTP_404_NOT_FOUND)
+
+        if not org.comstock_enabled or kwargs['pk'] != 'null':
+            return super(ColumnListingViewSet, self).retrieve(request, args, kwargs)
+
+        result = {
+            'status': 'success',
+            'data': {
+                'id': None,
+                'name': 'ComStock',
+                'settings_location': VIEW_LOCATION_TYPES[VIEW_LIST][1],
+                'inventory_type': VIEW_LIST_INVENTORY_TYPE[VIEW_LIST_PROPERTY][1],
+                'columns': []
+            }
+        }
+
+        comstock_columns = Column.objects.filter(organization_id=org_id, comstock_mapping__isnull=False) \
+            .order_by('comstock_mapping')
+
+        for index, column in enumerate(comstock_columns):
+            result['data']['columns'].append({
+                "id": column.id,
+                "pinned": False,
+                "order": index + 1,
+                "column_name": column.column_name,
+                "table_name": column.table_name,
+                "comstock_mapping": column.comstock_mapping
+            })
```

Comment:
I'm wondering if we could abstract out the logic of building out this ComStock setting record. I was envisioning that it could be reusable, but I could be missing something about how this is built in the `list` endpoint. 

My initial thought is that we could delegate it to the the ColumnListSetting model? But maybe it's not the model's "responsiblity". So maybe, in a utils file? Or maybe even just abstract it into a "_"/private method in this class?

What are your thoughts on this?

Reply:
Refactored into a single function 👍